concepts/pivoted/eICU-pivoted-OASIS.sql

```diff
@@ -0,0 +1,571 @@
+      WHEN hospitaladmitoffset BETWEEN (0.17*60) AND (4.94*60) THEN 3
+      WHEN hospitaladmitoffset BETWEEN (4.94*60) AND (24*60) THEN 0
+      WHEN hospitaladmitoffset BETWEEN (24.01*60) AND (311.80*60) THEN 2
+      WHEN hospitaladmitoffset > (311.80*60)  THEN 1
+      ELSE NULL
+      END AS pre_ICU_LOS_OASIS
+    FROM `physionet-data.eicu_crd.patient`
+    WHERE hospitaladmitoffset IS NOT NULL
+     AND patientunitstayid IS NOT NULL
+    GROUP BY pid_LOS, hospitaladmitoffset
+)
+
+-- Age 
+-- Change age from string to integer
+, age_numeric AS (
+  
+  SELECT patientunitstayid 
+  , CASE
+    WHEN age = "> 89" THEN 91
+    ELSE SAFE_CAST(age AS INT64) 
+    END AS age_num
+    FROM `physionet-data.eicu_crd.patient`
+    WHERE age IS NOT NULL
+      AND patientunitstayid IS NOT NULL
+    GROUP BY patientunitstayid, age
+)
+
+-- Get the information itself in a second step
+, age_OASIS AS (
+    SELECT patientunitstayid AS pid_age
+    , CASE
+    WHEN MAX(age_num < 24) THEN 0
+    WHEN MAX(age_num) BETWEEN 24 AND 53 THEN 3
+    WHEN MAX(age_num) BETWEEN 54 AND 77 THEN 6
+    WHEN MAX(age_num) BETWEEN 78 AND 89 THEN 9
+    WHEN MAX(age_num) > 89 THEN 7
+    ELSE NULL
+    END AS age_OASIS
+    FROM age_numeric
+    GROUP BY pid_age, age_num
+)
+
+-- GCS, Glasgow Coma Scale
+-- Merge information from two tables into one
+, merged_GCS AS (
+    
+    SELECT pivoted_gcs.patientunitstayid, pivoted_gcs.gcs
+    FROM `physionet-data.eicu_crd_derived.pivoted_gcs` AS pivoted_gcs
+   
+    FULL OUTER JOIN(
+      SELECT patientunitstayid, MIN(CAST(physicalexamvalue AS NUMERIC)) AS gcs
+      FROM `physionet-data.eicu_crd.physicalexam`
+      WHERE  (
+      (physicalExamPath LIKE "notes/Progress Notes/Physical Exam/Physical Exam/Neurologic/GCS/_" OR
+       physicalExamPath LIKE "notes/Progress Notes/Physical Exam/Physical Exam/Neurologic/GCS/__")
+      AND (physicalexamoffset > 0 AND physicalexamoffset <= 1440 ) -- consider only first 24h
+      AND physicalexamvalue IS NOT NULL
+      AND patientunitstayid IS NOT NULL
+      )
+  GROUP BY patientunitstayid
+  )
+  AS physicalexam
+  ON physicalexam.patientunitstayid = pivoted_gcs.patientunitstayid
+  WHERE (chartoffset > 0 AND chartoffset <= 1440) -- consider only first 24h
+    AND pivoted_gcs.gcs IS NOT NULL
+    AND pivoted_gcs.patientunitstayid IS NOT NULL 
+)
+
+
+-- Call merged_GCS table in one go
+, GCS_OASIS AS (
+    SELECT patientunitstayid AS pid_GCS
+    , CASE
+    WHEN MIN(gcs) < 8 THEN 10
+    WHEN MIN(gcs) BETWEEN 8 AND 13 THEN 4
+    WHEN MIN(gcs) = 14 THEN 3
+    WHEN MIN(gcs) = 15 THEN 0
+    ELSE NULL
+    END AS GCS_OASIS
+    FROM merged_GCS
+    --WHERE (chartoffset > 0 AND chartoffset <= 1440) -- already considered in step above
+    GROUP BY pid_GCS
+)
+
+-- Elective admission
+
+-- Mapping
+-- Assume emergency admission if patient came from
+-- Emergency Department, Direct Admit, Chest Pain Center, Other Hospital, Observation
```

Comment:
If you decide to augment the APACHE information, I would err on the side of being sensitive. That is, I would basically only include those admitted through the emergency department as non-elective. The rest could be elective admissions.

That being said, the APACHE covariate is probably sufficient for this!

Reply:
agrre, only picked ED admissions as non-elective